rs/embedders/tests/system_api.rs

```diff
@@ -2191,3 +2256,237 @@ fn get_system_api_for_best_effort_response(
+        ApiTypeBuilder::build_update_api(),
+        &SystemStateBuilder::default()
+            .environment_variables(env_vars)
+            .build(),
+        cycles_account_manager,
+    );
+
+    // Test ic0_env_var_count
+    assert_eq!(api.ic0_env_var_count().unwrap(), 2);
+
+    // Test ic0_env_var_name_size
+    assert_eq!(api.ic0_env_var_name_size(0).unwrap(), var_name_1.len()); // "TEST_VAR_1"
+    assert_eq!(api.ic0_env_var_name_size(1).unwrap(), var_name_2.len()); // "TEST_VAR_22"
+
+    // Test ic0_env_var_name_size with invalid index
+    assert!(matches!(
+        api.ic0_env_var_name_size(2),
+        Err(HypervisorError::EnvironmentVariableIndexOutOfBounds {
+            index: 2,
+            length: 2
+        })
+    ));
+
+    // Test ic0_env_var_name_copy
+    let mut heap = vec![0u8; 16];
+
+    // Copy first variable name
+    api.ic0_env_var_name_copy(0, 0, 0, 10, &mut heap).unwrap();
+    assert_eq!(&heap[0..10], var_name_1.as_bytes());
+
+    // Copy second variable name
+    api.ic0_env_var_name_copy(1, 0, 0, 11, &mut heap).unwrap();
+    assert_eq!(&heap[0..11], var_name_2.as_bytes());
+
+    // Test invalid index
+    assert!(matches!(
+        api.ic0_env_var_name_copy(2, 0, 0, 9, &mut heap),
+        Err(HypervisorError::EnvironmentVariableIndexOutOfBounds {
+            index: 2,
+            length: 2
+        })
+    ));
+
+    // Test invalid offset
+    assert!(matches!(
+        api.ic0_env_var_name_copy(0, 0, 10, 10, &mut heap),
+        Err(HypervisorError::ToolchainContractViolation { .. })
+    ));
+
+    // Test invalid size (destination buffer overflow)
+    assert!(matches!(
+        api.ic0_env_var_name_copy(0, 10, 0, 10, &mut heap),
+        Err(HypervisorError::ToolchainContractViolation { .. })
+    ));
+}
+
+// Helper function to copy test data to heap
+fn copy_to_heap(heap: &mut [u8], data: &[u8]) {
+    heap[..data.len()].copy_from_slice(data);
+}
+
+#[test]
+fn test_env_var_value_operations() {
+    let cycles_account_manager = CyclesAccountManagerBuilder::new().build();
+
+    let var_name_1 = "TEST_VAR_1".to_string();
+    let var_name_path = "PATH".to_string();
+    let var_name_empty = "EMPTY_VAR".to_string();
+    let var_value_1 = "Hello World".to_string();
+    let var_value_path = "/usr/local/bin:/usr/bin".to_string();
+
+    let mut env_vars = BTreeMap::new();
+    env_vars.insert(var_name_1.clone(), var_value_1.clone());
+    env_vars.insert(var_name_empty.clone(), "".to_string());
+    env_vars.insert(var_name_path.clone(), var_value_path.clone());
+
+    let api = get_system_api(
+        ApiTypeBuilder::build_update_api(),
+        &SystemStateBuilder::default()
+            .environment_variables(env_vars)
+            .build(),
+        cycles_account_manager,
+    );
+    let mut heap = vec![0u8; 256];
+
+    // Test ic0_env_var_count.
+    assert_eq!(api.ic0_env_var_count().unwrap(), 3);
+
+    // Test copying value for null bytes as variable name.
```

Comment:
This is also tested below requesting the name "NON_EXISTENT" so I'd drop one of those tests.